torch/csrc/jit/runtime/profiling_record.cpp

```diff
@@ -146,6 +164,43 @@ void ProfilingRecord::insertShapeProfile(Node* n, Value* i) {
   n->replaceInputWith(i, pn->output());
 }
 
+static void removeProfileNodes(Block* b) {
+  for (auto it = b->nodes().begin(); it != b->nodes().end(); it++) {
+    if (it->kind() == prim::profile) {
+      if (it->outputs().size() == 1) {
+        it->output()->replaceAllUsesWith(it->input(0));
+      }
+      it.destroyCurrent();
+    } else {
+      for (auto ib : it->blocks()) {
+        removeProfileNodes(ib);
+      }
+    }
+  }
+}
+
+void ProfilingRecord::peelLoops() {
+  // we peel starting out from the outermost loops, so we can avoid having
+  // peeled loops in outer loop peels
+  for (auto it = loops_with_profiled_nodes_.rbegin();
+       it != loops_with_profiled_nodes_.rend();
+       ++it) {
+    auto peel = PeelLoop(*it, 1);
```

Comment:
This is only ever invoked with `times=1`. So it makes sense toeither

* remove the times argument, and simplify the loop peeling `if(cond && trip_count > 0) { <first>; <orig_loop>; }`
* test that it actually works when times does not equal 1.

In addition, I think there are some tests that should be done checking this works for all combinations of present/absent loop conditions and max trip counts that are not max integer. Relying on existing tests seems dangerous.

Reply:
added tests